dbg.h

```diff
@@ -73,24 +73,63 @@ namespace pretty_function {
 #if defined(__clang__)
 #define DBG_MACRO_PRETTY_FUNCTION __PRETTY_FUNCTION__
 static constexpr size_t PREFIX_LENGTH =
-    sizeof("const char *dbg_macro::type_name_impl() [T = ") - 1;
+    sizeof("const char *dbg::type_name_impl() [T = ") - 1;
 static constexpr size_t SUFFIX_LENGTH = sizeof("]") - 1;
 #elif defined(__GNUC__) && !defined(__clang__)
 #define DBG_MACRO_PRETTY_FUNCTION __PRETTY_FUNCTION__
 static constexpr size_t PREFIX_LENGTH =
-    sizeof("const char* dbg_macro::type_name_impl() [with T = ") - 1;
+    sizeof("const char* dbg::type_name_impl() [with T = ") - 1;
 static constexpr size_t SUFFIX_LENGTH = sizeof("]") - 1;
 #elif defined(_MSC_VER)
 #define DBG_MACRO_PRETTY_FUNCTION __FUNCSIG__
 static constexpr size_t PREFIX_LENGTH =
-    sizeof("const char *__cdecl dbg_macro::type_name_impl<") - 1;
+    sizeof("const char *__cdecl dbg::type_name_impl<") - 1;
 static constexpr size_t SUFFIX_LENGTH = sizeof(">(void)") - 1;
 #else
 #error "This compiler is currently not supported by dbg_macro."
 #endif
 
 }  // namespace pretty_function
 
+// Formatting helpers
+
+template <typename T>
+struct print_formatted {
+  static_assert(std::is_integral<T>::value && std::is_unsigned<T>::value,
+                "Only unsigned integral types are supported.");
+
+  print_formatted(T value, int numeric_base)
+      : inner(value), base(numeric_base) {}
+
+  operator T() const {
+    return inner;
+  }
+
+  const char* prefix() const {
```

Comment:
Could it be better to use native `std::showbase` ?

https://en.cppreference.com/w/cpp/io/manip/showbase

Reply:
Only pitfall is that `std::oct << std::showbase << number` print `0<number>` instead of `0o<number>` as you did.

Reply:
I initially implemented it like this, but really didn't like the `0<number>` format, especially with the padding zeros.

Reply:
Great!